lib/objects.rb

```diff
@@ -396,6 +391,13 @@ def get_file_info(object, file, mdpi_barcode, comments)
       end
     end
 
+    # Get time offsets from structure. Use second segment if available, otherwise use first, then add 2 seconds.
+    begintimes = structure.xpath('//Span').collect{|d|d['begin']}
+    offset = structure_time_to_milliseconds(begintimes[[2,begintimes.count].min-1])
+    offset = offset + 2000
+    offset = 2000 if offset > file_hash[:files].collect { |d| d[:duration].to_i }.min
```

Comment:
Do we need to check if any file is shorter than 2sec? Maybe not for MDPI items.

Reply:
Good point.  Maybe I should make this:
```
calculated_duration = file_hash[:files].collect { |d| d[:duration].to_i }.min
offset = [2000, calculated_duration - 1000].min if offset > calculated_duration
```